cns/restserver/ipam.go

```diff
@@ -816,8 +823,15 @@ func (service *HTTPRestService) AssignAvailableIPConfigs(podInfo cns.PodInfo) ([
 
 	// Checks to make sure we found one IP for each NC
 	if len(ipsToAssign) != numOfNCs {
-		//nolint:goerr113 // return error
-		return podIPInfo, fmt.Errorf("not enough IPs available, waiting on Azure CNS to allocate more")
+		for _, ncID := range ncIDs {
+			if _, found := ipsToAssign[ncID]; found {
+				continue
+			}
+			if service.state.ContainerStatus[ncID].CreateNetworkContainerRequest.NCStatus == v1alpha.NCStatusSubnetFull {
+				return podIPInfo, errors.Errorf("not enough IPs available for %s, waiting on Azure CNS to allocate more but the Subnet is full", ncID)
```

Comment:
we should print the error code instead of expanding that in the error string, that way code gets updated all the way to containerd and cx can filter/take action on those error codes

Reply:
Printing the status code directly so that the data can go all the way to containerd.